File: app/src/main/java/com/eddief/android/wallofaww/model/ImageDao.java
package com.eddief.android.wallofaww.model;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import java.util.List;

@Dao
public interface ImageDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(AwwImage awwImage);

//    @Query("DELETE FROM aww_table")
//    void deleteAll();

    @Query("DELETE FROM aww_table WHERE primaryUrl LIKE :url")
    void remove(String url);

    @Query("SELECT * from aww_table")
    List<AwwImage> getSavedImages();

    @Query("SELECT * FROM aww_table WHERE primaryUrl LIKE :url")
    List<AwwImage> checkIfExists(String url);
}

File: app/src/main/java/com/eddief/android/infiniteaww/data/ImageDao.kt
package com.eddief.android.infiniteaww.data

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.eddief.android.infiniteaww.model.AwwImage

@Dao
interface ImageDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(awwImage: AwwImage)
    //    @Query("DELETE FROM aww_table")
    //    void deleteAll();
    @Query("DELETE FROM aww_table WHERE primaryUrl LIKE :url")
    fun remove(url: String)
    @Query("SELECT * from aww_table")
    fun getSavedImages(): List<AwwImage>
    @Query("SELECT * FROM aww_table WHERE primaryUrl LIKE :url")
    fun checkIfExists(url: String?): List<AwwImage>
}